middleware/http/oauth2clientcredentials/oauth2clientcredentials_middleware.go

```diff
@@ -0,0 +1,183 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package oauth2clientcredentials
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/dapr/components-contrib/middleware"
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/patrickmn/go-cache"
+	"github.com/valyala/fasthttp"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// Metadata is the oAuth clientcredentials middleware config
+type oAuth2ClientCredentialsMiddlewareMetadata struct {
+	ClientID            string `json:"clientID"`
+	ClientSecret        string `json:"clientSecret"`
+	Scopes              string `json:"scopes"`
+	TokenURL            string `json:"tokenURL"`
+	HeaderName          string `json:"headerName"`
+	EndpointParamsQuery string `json:"endpointParamsQuery,omitempty"`
+	AuthStyleString     string `json:"authStyle"`
+	AuthStyle           int    `json:"-"`
+}
+
+// dapr logger
+var log = logger.NewLogger("dapr.components-contrib.middleware.http.oauth2clientcredentials")
```

Comment:
@yaron2 , I have a question regarding the dapr logger, linter is complaining about the global variable here
```
middleware/http/oauth2clientcredentials/oauth2clientcredentials_middleware.go:39:5  gochecknoglobals  `log` is a global variable
```
So I tried to put the logger into the `Middleware` struct, but then there was no output in daprd?! Is that a known behavior of the logging mechanism or am I doing something wrong?

Reply:
It needs to be a struct member, you can look at other components. A logger is already injected from daprd

Reply:
https://github.com/dapr/components-contrib/blob/master/middleware/http/ratelimit/ratelimit_middleware.go

Reply:
Thanks, missed that I can pass the logger from daprd main.go!

Reply:
Changed, tested and pushed.

Reply:
The only drawback I see with that approach is that you can not identify which component logged a message or error. 

Scope will always be 'dapr.runtime' 
```
== DAPR == time="2020-08-13T04:53:00.2655148Z" level=debug msg="Cached token not found, try get one" app_id=msgraphapp instance=771becc56b7d scope=dapr.runtime type=log ver=edge
```